hikey_debian.mk

```diff
@@ -0,0 +1,408 @@
+ Trusted Applications.
+ NOTE! This package should only be used for testing and development.
+endef
+
+export CONTROL_TEXT
+
+.PHONY: deb
+deb: xtest optee-client
+	@mkdir -p $(DEBPKG_BIN_PATH) && cd $(DEBPKG_BIN_PATH) && \
+		cp -f $(OPTEE_CLIENT_EXPORT)/bin/tee-supplicant . && \
+		cp -f $(OPTEE_TEST_OUT_PATH)/xtest/xtest .
+
+	@mkdir -p $(DEBPKG_LIB_PATH) && cd $(DEBPKG_LIB_PATH) && \
+		cp $(OPTEE_CLIENT_EXPORT)/lib/libtee* .
+
+	@mkdir -p $(DEBPKG_TA_PATH) && cd $(DEBPKG_TA_PATH) && \
+		find $(OPTEE_TEST_OUT_PATH)/ta -name "*.ta" -exec cp {} . \;
+
+	@mkdir -p $(DEBPKG_CONTROL_PATH)
+	@echo "$$CONTROL_TEXT" > $(DEBPKG_CONTROL_PATH)/control
+	@cd $(OUT_PATH) && dpkg-deb --build optee_$(OPTEE_PKG_VERSION)
+
+################################################################################
+# Send built files to the host, note this require that the IP corresponds to
+# the device. One can run:
+#   IP=111.222.333.444 make send
+# If you don't want to edit the makefile itself.
+################################################################################
+.PHONY: send
+send:
+	@tar czf - $(shell cd $(OUT_PATH) && echo $(OUT_PATH)/*.deb && echo $(ROOT)/linux-image-*.deb) | ssh linaro@$(IP) "cd /tmp; tar xvzf -"
+	@echo "Files has been sent to $$IP/tmp/ and $$IP/tmp/out"
+	@echo "On the device, run:"
+	@echo " dpkg --force-all -i /tmp/out/optee_$(OPTEE_PKG_VERSION).deb"
+	@echo " dpkg --force-all -i /tmp/linux-image-*.deb"
+
+################################################################################
+# Flash
+################################################################################
+define flash_help
+	@read -r -p "1. Connect USB OTG cable, the micro USB cable (press any key)" dummy
+	@read -r -p "2. Connect HiKey to power up (press any key)" dummy
+endef
+
+.PHONY: recovery
+recovery:
+	@echo "Enter recovery mode to flash a new bootloader"
+	@echo "Jumper 1-2: Closed (Auto power up = Boot up when power is applied)"
+	@echo "       3-4: Closed (Boot Select = Recovery: program eMMC from USB OTG)"
+	$(call flash_help)
+	sudo python $(ROOT)/burn-boot/hisi-idt.py --img1=$(LLOADER_PATH)/l-loader.bin
+	@$(MAKE) --no-print flash FROM_RECOVERY=1
+
+.PHONY: flash
+flash:
+ifneq ($(FROM_RECOVERY),1)
+	@echo "Flash binaries using fastboot"
+	@echo "Jumper 1-2: Closed (Auto power up = Boot up when power is applied)"
+	@echo "       3-4: Open   (Boot Select = Boot from eMMC)"
+	@echo "       5-6: Closed (GPIO3-1 = Low: UEFI runs Fastboot app)"
+	$(call flash_help)
+	@echo "3. Wait until you see the (UART) message"
+	@echo "    \"Android Fastboot mode - version x.x Press any key to quit.\""
+	@read -r -p "   Then press any key to continue flashing" dummy
+endif
+	fastboot flash ptable $(LLOADER_PATH)/ptable-linux-4g.img
+	fastboot flash fastboot $(ARM_TF_PATH)/build/hikey/$(ARM_TF_BUILD)/fip.bin
+	fastboot flash nvme $(NVME_IMG)
+	fastboot flash boot $(BOOT_IMG)
+	fastboot flash system $(SYSTEM_IMG)
+
+.PHONY: flash-fip
+flash-fip:
```

Comment:
I understand that you'd like to have that here also, but I would like to keep this as a flash only target and "you need to know what to do before flashing". Even though it's nice to ensure that the binaries are in a good state, it's the same time rather annoying to see it jumping in and out of all sorts of gits when you just want to flash.

As a related side note. The "make run" target for QEMU etc. In the beginning that was rather quick and just checked a few well chosen files and just did the minimal amount of work to update the rootfs. Now, it seems like it's doing the same as "make all", which is kind of annoying I think, since it does a lot of stuff.

Summary: I won't add a dependency here :)